wgpu-core/src/validation.rs

```diff
@@ -321,6 +333,22 @@ pub enum StageError {
         var: InterfaceVar,
         limit: u32,
     },
+    #[error("Mesh shaders are limited to {limit} output vertices, but the shader has a maximum number of {value}")]
+    TooManyMeshVertices { limit: u32, value: u32 },
+    #[error("Mesh shaders are limited to {limit} output primitives, but the shader has a maximum number of {value}")]
+    TooManyMeshPrimitives { limit: u32, value: u32 },
+    #[error("Mesh or task shaders are limited to {limit} bytes of task payload, but the shader has a task payload of size {value}")]
+    TaskPayloadTooLarge { limit: u32, value: u32 },
+    #[error("Mesh shader's task payload has size {shader:?}, which doesn't match input from previous stage {input:?}")]
+    TaskPayloadMustMatch {
+        input: Option<u32>,
+        shader: Option<u32>,
+    },
+    #[error("Primitive index can only be used in a fragment shader if the preceding shader was a vertex shader or a mesh shader that writes to primitive index.
+    If a mesh shader writes to primitive index, it must be read by the fragment shader.")]
```

Comment:
```suggestion
    #[error("Primitive index can only be used in a fragment shader if the preceding shader was a vertex shader or a mesh shader that writes to primitive index. \
    If a mesh shader writes to primitive index, it must be read by the fragment shader.")]
```

Make sure you escape newlines in string literals.

This really needs to be two different errors. this is two separate situations: Fragment uses it, but geometry doesn't provide it; Mesh writes it, fragment doesn't consume it.

Reply:
I'll do that, but worth noting is that primitive index isn't even supported by naga yet. I just do this here so that when support lands it will work with mesh shdaders.